### ISSUE TITLE
skipping mount folder for permission denied

### ISSUE THREAD

**Issue by renyongzhe** [apptainer/singularity#5481]
### Version of Singularity:

```
3.4.2-1.1.el6
```

### Expected behavior

When mount multiple folder using -B , all would work well and the designated file would be found

### Actual behavior

some files in mounted folder could't be found.
When using debug mode, one folder was skipping moun for  permission denied.  Some key points list below.
What is the reason and how to solve it? 

```
WARNING [U=48,P=13804]     createLayer()                 skipping mount of /annovar/humandb: permission denied
DEBUG   [U=48,P=13804]     mountGeneric()                Mounting /annovar/humandb to /var/singularity/mnt/session/final/annovar/humandb
DEBUG   [U=48,P=13804]     mountGeneric()                Skipping mount, /var/singularity/mnt/session/final/annovar/humandb doesn't exist in container
```

### Steps to reproduce this behavior

` singularity -d exec -B /folder1,/folder2  gatk_4.1.2.0.sif perl example.pl  /folder2/testfile`

### What OS/distro are you running

```
Red Hat 4.4.7-18
```

### How did you install Singularity

may be source

**Comment by dtrudg** [comment]
Some mount issues have been addressed in Singularity versions newer than 3.4.2. Please update to a current version of Singularity and let us know if you have the same issue, and provide the full debug output if it occurs.

**Comment by renyongzhe** [comment]
> Some mount issues have been addressed in Singularity versions newer than 3.4.2. Please update to a current version of Singularity and let us know if you have the same issue, and provide the full debug output if it occurs.

**Comment by renyongzhe** [comment]
Thanks. I got the reason.  I have access to the specific diretory "human" but not the parent diretory , as a result mounting of singularity returned error permission denied.  Change the access  and all worked well.
 `chmod a+w /recurise_parent_diretory/`
